### PR TITLE
PAPP-25669: Convert this app to py3

### PR DESCRIPTION
**Pull request by paulnawat**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Converted this app to python 3 [PAPP-25669] 

## What is the current behavior? (OPTIONAL)
- This app is still in python2.

## What is the new behavior? (OPTIONAL)
- Converted this app to python 3 [PAPP-25669]. This app should be installable on `5.2.0` (i.e. the currently released Phantom version as of April 27)

## IMPORTANT
* The failed tests are expected, as this app is still a community app with lots of existing broken tests.
* We want to fix the failures incrementally to avoid reviewing a large PR.

## Tests
* The app can be installed on `5.3.1.84890`
image

---
Thanks for contributing!

### CODE REVIEW

File: exclude_files.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-*.py
```
Comment (paulnawat) [inline exclude_files.txt:1]: This is needed for the `xforce_connector.py` to be included in the install tar.

File: xforce_connector.py [hunk 2]
```diff
@@ -13,14 +13,17 @@
 # --
 
 import datetime
-from phantom.action_result import ActionResult
```
Comment (paulnawat) [inline xforce_connector.py:16]: The code change here is mostly suggested by the pre-commit runs.
Additional, 2to3 is run on this file.

File: xforce_connector.py [hunk 3]
```diff
@@ -146,12 +149,6 @@ def whois(self, param, action_id):
                 'Error running IBM X_Force WHOIS. Details:' + err.message
             )
 
-        if 'xforce_whois' not in whois_results:
-            return self.set_status_save_progress(
-                phantom.APP_ERROR,
-                'Error running IBM X_Force WHOIS. Details:' + err.message
```
Comment (paulnawat) [inline xforce_connector.py:152]: The err logic here is wrong, as the variable `err` should never be available here.

File: xforce.json [hunk 4]
```diff
@@ -1,18 +1,19 @@
 {
     "appid": "653a4735-1eed-4cb9-98c4-b40408e60b22",
     "name": "IBM XForce",
-    "description": "This app implements various <b>investigative</b> actions on the <b>IBM XForce</b> device",
+    "description": "This app implements various 'investigative' actions on the 'IBM XForce' device",
```
Comment (paulnawat) [inline xforce.json:4]: `Static tests` won't allow using `<b>` in the description field.

File: xforce.json [hunk 5]
```diff
@@ -44,13 +45,14 @@
     "actions": [
         {
             "action": "ip reputation",
-            "description": "Returns IP reputation report.",
+            "description": "Returns IP reputation report",
```
Comment (paulnawat) [inline xforce.json:48]: `Static tests` won't allow using `period` at the end of description.

File: xforce.json [hunk 6]
```diff
@@ -134,6 +136,20 @@
...
+                        1
+                    ]
+                },
+                {
+                    "data_path": "action_result.summary.total_objects_successful",
+                    "data_type": "numeric",
+                    "example_values": [
+                        1
+                    ]
+                },
```
Comment (paulnawat) [inline xforce.json:152]: These are the missing required path to get rid of install warnings.

**Comment by Paul-Kijtapart** [comment]
For reference, the `Static Tests` will be fixed in separate PRs to keep each one small and easy to review.

**Review by joshquerinphantom — APPROVED** [review]

**Pull request merged**